specutils/io/default_loaders/wcs_fits.py

```diff
@@ -171,17 +254,46 @@ def wcs1d_fits_writer(spectrum, file_name, hdu=0, update_header=False, **kwargs)
                        (isinstance(keyword[1], hdr_types) and
                         keyword[0] not in ('NAXIS', 'NAXIS1', 'NAXIS2'))])
 
-    # Cannot include uncertainty in IMAGE_HDU - maybe provide option to
-    # separately write this to BINARY_TBL extension later.
-    if spectrum.uncertainty is not None:
-        warnings.warn("Saving uncertainties in wcs1d format is not yet supported!",
-                      AstropyUserWarning)
-
     # Add flux array and unit
     ftype = kwargs.pop('dtype', spectrum.flux.dtype)
     funit = u.Unit(kwargs.pop('unit', spectrum.flux.unit))
-    flux = spectrum.flux.to(funit, equivalencies=u.spectral_density(wl))
+    flux = spectrum.flux.to(funit, equivalencies=u.spectral_density(disp))
     hdulist[0].data = flux.value.astype(ftype)
+    if flux_name is not None:
+        hdulist[0].name = flux_name
+
+    # Append mask array (duplicate WCS for that extension)?
```

Comment:
Usually mask only makes sense in pixel space. Putting WCS on that extension would be misleading. In HST images, I think the WCS is only in the SCI extensions.

Reply:
Yes, looking at the Mamba format I don't see any WCS info in MASK or IVAR; was just wondering what might happen when something is reading those HDUs separately, finding celestial axes without any coordinates.